pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -374,6 +385,59 @@ func (r *ReconcileStorageCluster) ensureCephObjectStores(instance *ocsv1.Storage
 	return nil
 }
 
+// newExternalCephObjectStoreInstances returns a set of CephObjectStores
+// needed for external cluster mode
+func (r *ReconcileStorageCluster) newExternalCephObjectStoreInstances(
+	initData *ocsv1.StorageCluster, rgwEndpoint string, reqLogger logr.Logger) ([]*cephv1.CephObjectStore, error) {
+	// check whether the provided rgw endpoint is empty
+	if rgwEndpoint = strings.TrimSpace(rgwEndpoint); rgwEndpoint == "" {
+		reqLogger.Info("WARNING: Empty RGW Endpoint specified, external CephObjectStore won't be created")
+		return nil, nil
+	}
+
+	var gateWay cephv1.GatewaySpec
```

Comment:
What about a method that returns a `cephv1.GatewaySpec` instead of this long method? It will be easier to unit test too.

Reply:
Done...